packages/back-end/src/models/SdkConnectionModel.ts

```diff
@@ -250,6 +252,14 @@ export async function editSDKConnection(
     }
   );
 
+  const payloadKey: SDKPayloadKey = {
+    environment: connection.environment,
+    project: connection.project,
+  };
```

Comment:
This will purge more things than necessary.  If two SDK connections have the same environment/project, this will purge both of them, even though only one of them has changed.

When we cache features, we set 3 surrogate keys and any of them can be used to purge the cache.
 https://github.com/growthbook/growthbook/blob/main/packages/back-end/src/controllers/features.ts#L190
- The organization id
- The SDK Connection key
- The payload key (combo of environment and project)

When features change, we need to use the payload key since there may be multiple SDK connections that are affected and we want to purge all of them at once.

In this case, only a single SDK connection is changing, so we can use `connection.key` instead in the `surrogate-key` header.

Reply:
Thanks for this comment! That makes a lot of sense and I just addressed it. Is it worth renaming `getSurrogateKey` to `getSurrogateKeyForPayloadKey` or something more descriptive? I think looking at the function initially, I thought all surrogate keys would be formed using the ``${orgId}_${project}_${environment}`` format